rules/set-gmsacredentialspecname-value/raw.rego

```diff
@@ -0,0 +1,45 @@
+package armo_builtins
+
+import future.keywords.if
+
+# Rule to check if the GMSA credential spec name is set in the security context
+deny[msga] {
+    # Verify the object kind
+    obj := input[_]
+    allowed_kinds := {"Pod", "Deployment", "ReplicaSet", "DaemonSet", "StatefulSet", "Job", "CronJob"}
+    allowed_kinds[obj.kind]
+
+    # Check if the GMSA credential spec name is set in the security context
+    not gmsaCredentialSpecNameSet(getSecurityContext(obj))
+
+    path := sprintf("%v.securityContext.gmsaCredentialSpecName", [obj.kind])
+    msga := {
+        "alertMessage": sprintf("%v: %v does not set 'securityContext.gmsaCredentialSpecName' with allowed value", [obj.kind, obj.metadata.name]),
+        "packagename": "armo_builtins",
+        "alertScore": 7,
+        "failedPaths": [path],
+        "fixPaths": [],
+        "alertObject": {
+            "k8sApiObjects": [obj]
+        }
+    }
+}
+
+# Function to check if the GMSA credential spec name is set in the security context
+gmsaCredentialSpecNameSet(securityContext) := true if {
+    securityContext != null
+    securityContext.windowsOptions != null
+    securityContext.windowsOptions.gmsaCredentialSpecName != null
+} else := false
+
+# Function to get the security context of an object
+getSecurityContext(obj) = obj.spec.securityContext {
```

Comment:
Despite the function seems to work properly, we should keep the format used in other policies.
So, usually we split the policy in 3 rules (`Cronjob`, `Pod` and `Workload`, where `Workload` includes `Deployment`, `ReplicaSet`, `DaemonSet`, `StatefulSet`).
Here's an example: https://github.com/kubescape/regolibrary/blob/master/rules/set-seLinuxOptions/raw.rego